server/app/assets/javascripts/north_star_modal.ts

```diff
@@ -0,0 +1,71 @@
+export class NorthStarModalController {
+  /**
+   * Find the modals, and add on-click listeners on their respective buttons to toggle them.
+   * @param {Element} modalContainer The container holding the modal.
+   * @param {HTMLElement} modal The modal element.
+   **/
+  static attachModalListener(modal: HTMLElement) {
+    // When the button is clicked to trigger the modal, save to local storage that that modal type
+    // has been shown.
+    const onlyShowOnceGroup = modal.getAttribute('only-show-once-group')
+    // We use dialog-id instead of the id attribute because USWDS does some DOM manipulation and
+    // moves the id to a wrapper element, so it no longer points to the dialog itself.
+    const modalButtons = document.querySelectorAll(
+      `a[href='#` + modal.getAttribute('dialog-id') + `']`,
+    )
+    if (onlyShowOnceGroup) {
+      modalButtons.forEach((modalButton) => {
+        modalButton.addEventListener('click', () => {
+          localStorage.setItem(onlyShowOnceGroup, 'true')
+          NorthStarModalController.maybeUpdateButtonHrefToBypassUrl(modal)
+        })
+      })
+    }
+  }
+
+  static maybeUpdateButtonHrefToBypassUrl(modal: HTMLElement) {
+    const onlyShowOnceGroup = modal.getAttribute('only-show-once-group')
+    if (onlyShowOnceGroup) {
+      const modalHasBeenShown = localStorage.getItem(onlyShowOnceGroup)
```

Comment:
question (non-blocking): the way this works now, this will only ever be true for `login-prompt` right? do all modal types use the `ModalFragment`?

Reply:
for right now yes. In J2HTML there are two only-show-once-groups: 1 for login dialogs in the program index page and another in the review page when the user is redirected from a program slug, but I haven't implemented the latter yet

Reply:
and all modals in the northstar work will use modalfragment. there will be a little bit of refactoring required for that template to ensure the right ones will be present on the right pages as I build out all of the available modals